homeassistant/components/demo/mailbox.py

```diff
@@ -1,17 +1,25 @@
 """Support for a demo mailbox."""
+from __future__ import annotations
+
 from hashlib import sha1
 import logging
 import os
 
 from homeassistant.components.mailbox import CONTENT_TYPE_MPEG, Mailbox, StreamError
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
 from homeassistant.util import dt
 
 _LOGGER = logging.getLogger(__name__)
 
 MAILBOX_NAME = "DemoMailbox"
 
 
-async def async_get_handler(hass, config, discovery_info=None):
+async def async_get_handler(
+    hass: HomeAssistant,
+    config: ConfigType,
+    discovery_info: DiscoveryInfoType | None = None,
+) -> Mailbox | None:
```

Comment:
This won't return `None`?

```suggestion
) -> Mailbox:
```

Reply:
Unless I am mistaken, the base class allows None, so I think it is better to standardise accross all components that implement this method.

Reply:
I think it is better to represent/type what is at hand.
Typing should be tailored against the expected use of this method. For this method, it is expected to return `Mailbox`, not `None`.

Doing so, still makes the signature compatible with the parent.

Reply:
Ok. Done